libraries/core_libs/consensus/src/dag/dag_block_manager.cpp

```diff
@@ -221,10 +222,30 @@ DagBlockManager::InsertAndVerifyBlockReturnType DagBlockManager::verifyBlock(con
     markBlockInvalid(block_hash);
     return InsertAndVerifyBlockReturnType::NotEligible;
   }
+  {
+    u256 total_block_weight = 0;
+    const auto &trxs = blk.getTrxs();
+    const auto &trxs_gas_estimations = blk.getTrxsGasEstimations();
+    if (trxs.size() != trxs_gas_estimations.size()) {
+      return InsertAndVerifyBlockReturnType::IncorrectTransactionsEstimation;
+    }
+    final_chain_->wait_for_finalization(*propose_period, stopped_);
```

Comment:
This will cause stuck here until execute to proposed period.
I think that's better for mark the block for FutureBlock, and process it again later. Please see above line 212 - 217

Reply:
Oh. So this waits doesn't make sense because we are already implicitly checking this in `dpos_is_eligible` method. I will just remove it and add comment